op-challenger/game/service.go

```diff
@@ -73,10 +73,15 @@ func NewService(ctx context.Context, logger log.Logger, cfg *config.Config) (*Se
 		disk,
 		cfg.MaxConcurrency,
 		func(addr common.Address, dir string) (scheduler.GamePlayer, error) {
-			return fault.NewGamePlayer(ctx, logger, m, cfg, dir, addr, txMgr, client)
+			return fault.NewGamePlayer(ctx, logger, m, cfg, dir, addr, txMgr, l1Client)
 		})
 
-	monitor := newGameMonitor(logger, cl, loader, sched, cfg.GameWindow, client.BlockNumber, cfg.GameAllowlist)
+	polledClient, err := client.DialPolledClientWithTimeout(ctx, client.DefaultDialTimeout, logger, cfg.L1EthRpc, cfg.PollInterval)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial L1: %w", err)
+	}
+
+	monitor := newGameMonitor(logger, cl, loader, sched, cfg.GameWindow, l1Client.BlockNumber, cfg.GameAllowlist, polledClient)
```

Comment:
It's really unpleasant to have to create a completely separate connection to the L1 to support monitoring.  I think we can avoid it by reusing the existing RPC connection and just wrapping when needed though:
```suggestion
	var rpcClient client.RPC = client2.NewBaseRPCClient(l1Client.Client())
	if regexp.MustCompile("^http(s)?://").MatchString(cfg.L1EthRpc) {
		rpcClient = client2.NewPollingClient(ctx, logger, rpcClient, client2.WithPollRate(cfg.PollInterval))
	}
	monitor := newGameMonitor(logger, cl, loader, sched, cfg.GameWindow, l1Client.BlockNumber, cfg.GameAllowlist, rpcClient)
```

That's basically duplicating logic from https://github.com/ethereum-optimism/optimism/blob/392dbddeb6f4516d2e7759fa162bba7b58addff5/op-node/client/rpc.go#L91-L101 so probably worth just extracting that as a new method in rpc.go but hopefully this gives the basic idea.